residual plots for DQM

aspiezia opened the pull request:
Mean and RMS of the residuals of the tracks have been added in the DQM. Details can be found in this presentation:
https://indico.cern.ch/event/536893/contributions/2389697/attachments/1388650/2114441/AnielloSpiezia_141216.pdf

aspiezia committed:
residual plots for DQM

aspiezia committed:
residual plots for DQM, remove useless histos

aspiezia committed:
residual plots for DQM, remove useless histos

cmsbuild commented:
A new Pull Request was created by  @aspiezia  for CMSSW_9_0_X.

It involves the following packages:

DQM/SiStripMonitorClient
DQM/TrackerMonitorTrack


@cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@hdelanno, @makortel, @fioriNTU, @idebruyn, @threus, @venturia this is something you requested to watch as well.
@davidlange6, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_9_0_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

cmsbuild commented:
+1
Tested at: a0a714e6c65870a27ee5a346d1d0e75367573276
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17645/18009/summary.html

cmsbuild commented:
Comparison job queued.

davidlange6 commented:
-1
pending answers to comment

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17645/18009/summary.html

boudoul commented:
Hi @aspiezia , could you please adress the comments ? Thanks

fioriNTU commented:
@boudul me and @aspiezia already plan to look into it tomorrow, he is busy with an analysis approval at the moment. Will be back to you soon.

aspiezia commented on DQM/TrackerMonitorTrack/src/MonitorTrackResiduals.cc:
```diff
@@ -38,6 +38,11 @@ void MonitorTrackResidualsBase<pixel_or_strip>::bookHistograms(DQMStore::IBooker
+  std::string topFolderName_ = "SiStrip";
+  SiStripFolderOrganizer folder_organizer;
+  folder_organizer.setSiStripFolderName(topFolderName_);
+  tkhisto_ResidualsMean = new TkHistoMap(ibooker , topFolderName_ ,"TkHMap_ResidualsMean", 0.0,true);
```
HI, I cannot find the end of job where I can delete the object. I have tried to look for some examples in cmssw and I cannot find a place where these tkhisto maps are deleted (see for example https://github.com/aspiezia/cmssw/blob/a0a714e6c65870a27ee5a346d1d0e75367573276/DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc#L153). Do you have a suggestion?

dmitrijus replied:
Hi @aspiezia,
forget about deleting at the end job and do what @davidlange6 suggested - use unique_ptrs.

Modify https://github.com/aspiezia/cmssw/blob/a0a714e6c65870a27ee5a346d1d0e75367573276/DQM/SiStripMonitorTrack/interface/SiStripMonitorTrack.h#L131 and change TkHistoMap *x to std::unique_ptr<TkHistoMap> x;

aspiezia replied:
Hi @dmitrijus and @davidlange6, thanks a lot for the suggestion. 

Consider that https://github.com/aspiezia/cmssw/blob/a0a714e6c65870a27ee5a346d1d0e75367573276/DQM/SiStripMonitorTrack/interface/SiStripMonitorTrack.h#L131 is not the code I am modifying. It was just an example showing that everywhere in cmssw, the tracker maps are defined as I am doing. 

Anyway, I have tried your suggestion in the code I am modifing and it does not compile because of this line: https://github.com/aspiezia/cmssw/blob/a0a714e6c65870a27ee5a346d1d0e75367573276/DQM/SiStripMonitorTrack/interface/SiStripMonitorTrack.h#L131 
Do you have a suggestion on how to change it?

Thanks,
Aniello

aspiezia committed:
remove double tracker map

cmsbuild commented:
Pull request #17645 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

aspiezia commented:
I have removed the "TkHMap_ResidualsRMS" that was a copy of "TkHMap_ResidualsMean" and I have added the RMS map in DQM/SiStripMonitorClient/src/SiStripTrackerMapCreator.cc as suggested by @fioriNTU 
I am still waiting for a suggestion from @davidlange6 to understand how to delete the object tracker map at the end of the job.
Aniello

davidlange6 commented:
i'm not sure I understand the question? You have some choices- 

a)if you need to use bare pointers for some reason (I can't think of one), then anything you new has to be deleted when you are done with the object. 
b) use a unique_ptr instead and then the compiler will manage the memory for you
c) don't use a pointer. Why is it needed in this case?

davidlange6 commented:
> On Mar 8, 2017, at 11:50 AM, aspiezia <notifications@github.com> wrote:
> 
> @aspiezia commented on this pull request.
> 
> In DQM/TrackerMonitorTrack/src/MonitorTrackResiduals.cc:
> 
> > @@ -38,6 +38,11 @@ void MonitorTrackResidualsBase<pixel_or_strip>::bookHistograms(DQMStore::IBooker
>      m_cacheID_ = cacheID;
>      this->createMEs( ibooker , iSetup);
>    }
> +  std::string topFolderName_ = "SiStrip";
> +  SiStripFolderOrganizer folder_organizer;
> +  folder_organizer.setSiStripFolderName(topFolderName_);
> +  tkhisto_ResidualsMean = new TkHistoMap(ibooker , topFolderName_ ,"TkHMap_ResidualsMean", 0.0,true);
> 
> Hi @dmitrijus and @davidlange6, thanks a lot for the suggestion.
> 
> Consider that https://github.com/aspiezia/cmssw/blob/a0a714e6c65870a27ee5a346d1d0e75367573276/DQM/SiStripMonitorTrack/interface/SiStripMonitorTrack.h#L131 is not the code I am modifying. It was just an example showing that everywhere in cmssw, the tracker maps are defined as I am doing.
> 

Right, looks like an old bug that should be addressed

> Anyway, I have tried your suggestion in the code I am modifing and it does not compile because of this line: https://github.com/aspiezia/cmssw/blob/a0a714e6c65870a27ee5a346d1d0e75367573276/DQM/SiStripMonitorTrack/interface/SiStripMonitorTrack.h#L131
> Do you have a suggestion on how to change it?

I think your link is incorrect?

> 
> Thanks,
> Aniello
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

aspiezia commented:
Right @davidlange6,
this is the correct link:
https://github.com/aspiezia/cmssw/blob/86068d7740011458e797555f7ef745502f349271/DQM/TrackerMonitorTrack/src/MonitorTrackResiduals.cc#L44
Aniello

davidlange6 commented:
yes, you need to follow the right syntax (easy to ask google) - something like

tkhisto_ResidualsMean = std::unique_ptr<TkHistoMap>(new TkHistoMap(ibooker , topFolderName_ ,"TkHMap_ResidualsMean", 0.0,true));

> On Mar 8, 2017, at 12:01 PM, aspiezia <notifications@github.com> wrote:
> 
> Right @davidlange6,
> this is the correct link:
> https://github.com/aspiezia/cmssw/blob/86068d7740011458e797555f7ef745502f349271/DQM/TrackerMonitorTrack/src/MonitorTrackResiduals.cc#L44
> Aniello
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

aspiezia committed:
TkHistoMap to std::unique_ptr<TkHistoMap>

cmsbuild commented:
Pull request #17645 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

aspiezia commented:
Thanks a lot for the suggestion. The change has been implemented.

aspiezia committed:
ResidualsRMS to ResidualsRMS_

cmsbuild commented:
Pull request #17645 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_9_0_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

cmsbuild commented:
+1
Tested at: b3214f8faef4ab80752a1f7932006e8e7f5b2166
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17645/18603/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17645/18603/summary.html

davidlange6 commented:
+1
